Patch to LB deployment modul (with build workarounds, for using).

Index: org.glite.deployment.lb/config/scripts/glite-lb-config.py
===================================================================
RCS file: /cvs/jra1mw/org.glite.deployment.lb/config/scripts/glite-lb-config.py,v
retrieving revision 1.63.2.8
diff -u -r1.63.2.8 glite-lb-config.py
--- org.glite.deployment.lb/config/scripts/glite-lb-config.py	22 Feb 2006 00:12:59 -0000	1.63.2.8
+++ org.glite.deployment.lb/config/scripts/glite-lb-config.py	3 Apr 2006 18:59:49 -0000
@@ -120,7 +120,12 @@
 
         if not os.path.exists('/tmp/mysql.sock'):
             os.symlink('/var/lib/mysql/mysql.sock', '/tmp/mysql.sock')
-            
+        
+
+        #-------------------------------------------------------------------
+        # start bkserver
+        #-------------------------------------------------------------------
+
         pid = glib.getPID('bkserverd')
         if pid != 0:
             print 'The gLite LB Server service is already running. Restarting...'
@@ -133,7 +138,7 @@
         pid = glib.getPID('bkserverd')
 
         if (pid != 0):
-            print "The gLite LB Server service has been started               ",
+            print "The gLite LB Server service has been started              ",
             glib.printOkMessage()
         else:
             glib.printErrorMessage("Could not start the gLite LB Server service")
@@ -142,6 +147,36 @@
             return 1
         
         #-------------------------------------------------------------------
+        # start jp-importer, if enabled
+        #-------------------------------------------------------------------
+
+        lb_export = 0
+        if params.has_key('lb.export.enabled'):
+            if params['lb.export.enabled'] == "true":
+                lb_export = 1
+
+        if lb_export:
+            pid = glib.getPID('jp-importer')
+            if pid != 0:
+                print 'The gLite JP Importer service is already running. Restarting...'
+                os.system('%s/etc/init.d/glite-jp-importer stop' % os.environ['GLITE_LOCATION'])
+            else:
+                print 'Starting the gLite JP Importer service...'
+
+            os.system('%s/etc/init.d/glite-jp-importer start' % os.environ['GLITE_LOCATION'])
+
+            pid = glib.getPID('jp-importer')
+
+            if (pid != 0):
+                print "The gLite JP Importer service has been started            ",
+                glib.printOkMessage()
+            else:
+                glib.printErrorMessage("Could not start the gLite JP Importer service")
+                glib.printErrorMessage("Please verify and re-run the script                        "),
+                glib.printFailedMessage()
+                return 1
+        
+        #-------------------------------------------------------------------
         # Start Servicetool
         #-------------------------------------------------------------------
 
@@ -161,6 +196,10 @@
         if (pid != 0):
             os.system('%s/etc/init.d/glite-lb-bkserverd stop' % os.environ['GLITE_LOCATION'])
 
+        #-------------------------------------------------------------------
+        # Book Keeping Server
+        #-------------------------------------------------------------------
+
         pid = glib.getPID('bkserverd')
         if (pid != 0):
             print 'Could not stop the LB Server service            ',
@@ -171,6 +210,25 @@
             glib.printOkMessage()
         
         #-------------------------------------------------------------------
+        # JP Importer
+        #-------------------------------------------------------------------
+
+        pid = glib.getPID('jp-importer')
+        if (pid != 0):
+            os.system('%s/etc/init.d/glite-jp-importer stop' % os.environ['GLITE_LOCATION'])
+
+        pid = glib.getPID('jp-importer')
+        if (pid != 0):
+            print 'Could not stop the JP Importer service                  ',
+            glib.printFailedMessage()
+            error_level = 1
+        else:
+            if params.has_key('lb.export.enabled'):
+                if params['lb.export.enabled'] == "true":
+                    print 'The JP Importer service has been stopped                  ',
+                    glib.printOkMessage()
+        
+        #-------------------------------------------------------------------
         # MySQL
         #-------------------------------------------------------------------
 
@@ -195,6 +253,10 @@
         if retval != 0:
             error_level = 1
 
+        retval = os.system('%s/etc/init.d/glite-jp-importer status' % os.environ['GLITE_LOCATION'])
+        if retval != 0:
+            error_level = 1
+
         #-------------------------------------------------------------------
         # Servicetool
         #-------------------------------------------------------------------
@@ -325,7 +387,29 @@
             print "\n==> MySQL database %s already exist\n" % params['lb.database.name']
             
         self.mysql.stop()
-    
+
+        # ------------------------------------------------------------------
+        # export from bkserver to cron
+        # ------------------------------------------------------------------
+	if params['GLITE_LB_EXPORT_ENABLED'] == "true":
+	    file = open('%s/etc/glite-lb-export-cron-wrapper.sh' % os.environ['GLITE_LOCATION'], 'w')
+	    file.write('#! /bin/sh\n')
+	    file.write('. %s\n' % glib.getInstallerExportFile())
+            file.write('$GLITE_LOCATION/sbin/glite-lb-export.sh\n')
+	    file.close()
+            os.system('/bin/chmod 0755 %s/etc/glite-lb-export-cron-wrapper.sh' % os.environ['GLITE_LOCATION'])
+
+	    file = open('/etc/cron.d/glite-lb-export.cron', 'w')
+	    file.write('# periodically run purge and export jobs from bkserver\n')
+	    file.write('00,30 * * * * %s %s/etc/glite-lb-export-cron-wrapper.sh\n' % (params['glite.user.name'], os.environ['GLITE_LOCATION']))
+	    file.close()
+            os.system('/bin/chmod 0755 /etc/cron.d/glite-lb-export.cron')
+	else:
+            os.system('/bin/rm -f /etc/cron.d/glite-lb-export.cron')
+
+            # Touch cron spool directory to cause reloading of the crontabs
+            os.system("/bin/touch /var/spool/cron")
+
         #-------------------------------------------------------------------
         # RGMA servicetool: configure servicetool
         #-------------------------------------------------------------------
@@ -411,6 +495,17 @@
     # Perl
     glib.addEnvPath("PERL5LIB", "%s/lib/perl:%s/lib/perl5" % (os.environ['GPT_LOCATION'],os.environ['GLITE_LOCATION']))
 
+    # LB export
+    glib.export('GLITE_LB_EXPORT_ENABLED', params['lb.export.enabled']);
+    glib.export('GLITE_LB_EXPORT_JPPS', params['lb.export.jpps']);
+    glib.export('GLITE_LB_EXPORT_JPREG_MAILDIR', params['lb.export.jpreg']);
+    glib.export('GLITE_LB_EXPORT_JPDUMP_MAILDIR', params['lb.export.jpdump']);
+    glib.export('GLITE_LB_EXPORT_DUMPDIR', params['lb.export.dump']);
+    glib.export('GLITE_LB_EXPORT_DUMPDIR_KEEP', params['lb.export.dump.keep']);
+    glib.export('GLITE_LB_EXPORT_JOBSDIR', params['lb.export.jobs']);
+    glib.export('GLITE_LB_EXPORT_JOBSDIR_KEEP', params['lb.export.jobs.keep']);
+    glib.export('GLITE_LB_EXPORT_PURGE_ARGS', '"%s"' % params['lb.export.purgeargs']);
+
     # Set environment
     glib.setUserEnv()
     
Index: org.glite.deployment.lb/config/templates/glite-lb.cfg.xml
===================================================================
RCS file: /cvs/jra1mw/org.glite.deployment.lb/config/templates/glite-lb.cfg.xml,v
retrieving revision 1.20.2.5
diff -u -r1.20.2.5 glite-lb.cfg.xml
--- org.glite.deployment.lb/config/templates/glite-lb.cfg.xml	22 Feb 2006 00:12:58 -0000	1.20.2.5
+++ org.glite.deployment.lb/config/templates/glite-lb.cfg.xml	3 Apr 2006 18:59:49 -0000
@@ -17,9 +17,16 @@
 			parameter. Leave it empty of comment it out to use the same as 'glite.user.name'"
 			value="changeme"/>
              	
-	    <mysql.root.password
-	        description="The mysql root password"
-	        value="changeme"/>
+		<mysql.root.password
+		        description="The mysql root password"
+	        	value="changeme"/>
+
+		<lb.export.jpps
+			description="Job Provenance Primary Storage.
+				[Example: localhost:8901][Type: string]"
+			value="changeme"/>
+
+
 	
 		<!-- xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx -->
 		<!-- Advanced parameters - Change them if you know what you're doing -->
@@ -55,6 +62,12 @@
 				[Example: 17M][Type: Integer][Unit: MB]"
 			value="17M"/>
 
+		<lb.export.enabled
+			description="Enables exports to Job Provenance.
+				[Example: true][Type: boolean]"
+			value="true"/>
+
+
 		<!-- xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx -->
 		<!-- System parameters - You should leave these alone -->
 		<!-- xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx -->		
@@ -67,7 +80,42 @@
 			<value>destination</value>
 
 		</lb.index.list>
-			
+
+		<lb.export.jpreg
+			description="Spool directory where job registration requests are stored before delivered to JPPS.
+				[Type: string]"
+			value="/tmp/jpreg"/>
+
+		<lb.export.jpdump
+			description="Spool directory where dump upload requests are stored before delivered to JPPS.
+				[Type: string]"
+			value="/tmp/jpdump"/>
+
+		<lb.export.dump
+			description="Dump directory for purged jobs.
+				[Type: string]"
+			value="/tmp/dump"/>
+
+		<lb.export.dump.keep
+			description="Dump directory for handled purged jobs.
+				[Example: /tmp/dump.keep][Type: string]"
+			value=""/>
+
+		<lb.export.jobs
+			description="Exported events divided to file per job before delivered to JPPS.
+				[Type: string]"
+			value="/tmp/lb_export"/>
+
+		<lb.export.jobs.keep
+			description="Keep the exported events.
+				[Example: /tmp/lb_export.keep][Type: string]"
+			value=""/>
+
+		<lb.export.purgeargs
+			description="Purge arguments (with timeouts).
+				[Example: -a 1h -c 1h -n 1h -o 1d][Type: string]"
+			value="--aborted 1d --cleared 1d --cancelled 1d --other 7d"/>
+
 	</parameters>
 
 </config>
Index: org.glite.deployment.lb/project/glite-lb.sdf.xml.template
===================================================================
RCS file: /cvs/jra1mw/org.glite.deployment.lb/project/glite-lb.sdf.xml.template,v
retrieving revision 1.37.2.3
diff -u -r1.37.2.3 glite-lb.sdf.xml.template
--- org.glite.deployment.lb/project/glite-lb.sdf.xml.template	8 Jan 2006 16:44:58 -0000	1.37.2.3
+++ org.glite.deployment.lb/project/glite-lb.sdf.xml.template	3 Apr 2006 18:59:49 -0000
@@ -57,6 +57,12 @@
 					build="@org.glite.lb.server-bones.info.build@"
 					arch="i386"/>
 					   
+				<component name="glite-lb-client"
+					version="@org.glite.lb.client.info.version@"
+					age="@org.glite.lb.client.info.age@"
+					build="@org.glite.lb.client.info.build@"
+					arch="i386"/>
+					   
 				<component name="glite-wms-utils-jobid"
 					version="@org.glite.wms-utils.jobid.info.version@"
 					age="@org.glite.wms-utils.jobid.info.age@"
@@ -105,6 +111,12 @@
                     build="@org.glite.jp.ws-interface.info.build@"
                     arch="i386"/>
 
+				<component name="glite-jp-client"
+                    version="@org.glite.jp.client.info.version@"
+                    age="@org.glite.jp.client.info.age@"
+                    build="@org.glite.jp.client.info.build@"
+                    arch="i386"/>
+
                 <component name="gridsite"
                     version="@org.gridsite.core.info.version@"
                     age="@org.gridsite.core.info.age@"
Index: org.glite.deployment.lb/project/properties.xml
===================================================================
RCS file: /cvs/jra1mw/org.glite.deployment.lb/project/properties.xml,v
retrieving revision 1.2.4.1
diff -u -r1.2.4.1 properties.xml
--- org.glite.deployment.lb/project/properties.xml	27 Nov 2005 16:16:17 -0000	1.2.4.1
+++ org.glite.deployment.lb/project/properties.xml	3 Apr 2006 18:59:49 -0000
@@ -65,5 +65,7 @@
 %attr(644,root,root) %{prefix}/share/doc/glite-lb/release_notes/release_notes.pdf\n
 %attr(644,root,root) %{prefix}/share/doc/glite-lb/release_notes/release_notes.html\n"
 	/>
+	<property name="build.package.name" value="glite-lb-config" />
+	<property name="build.package.prefix" value="glite-lb" />
 
 </project>
Index: org.glite.deployment.rgma-servicetool/project/properties.xml
===================================================================
RCS file: /cvs/jra1mw/org.glite.deployment.rgma-servicetool/project/properties.xml,v
retrieving revision 1.3.4.1
diff -u -r1.3.4.1 properties.xml
--- org.glite.deployment.rgma-servicetool/project/properties.xml	17 Nov 2005 16:15:16 -0000	1.3.4.1
+++ org.glite.deployment.rgma-servicetool/project/properties.xml	3 Apr 2006 18:59:49 -0000
@@ -71,5 +71,7 @@
 %attr(644,root,root) %{prefix}/share/doc/glite-rgma-servicetool/release_notes/release_notes.doc\n
 %attr(644,root,root) %{prefix}/share/doc/glite-rgma-servicetool/release_notes/release_notes.pdf\n
 %attr(644,root,root) %{prefix}/share/doc/glite-rgma-servicetool/release_notes/release_notes.html" />
-		 
+
+<property name="build.package.name" value="glite-rgma-servicetool-config" />
+<property name="build.package.prefix" value="glite-rgma-servicetool" />
 </project>
